client.go

```diff
@@ -873,6 +873,7 @@ func (c *Client[TTx]) Start(ctx context.Context) error {
 
 		for _, producer := range c.producersByQueueName {
 			if err := producer.StartWorkContext(fetchCtx, workCtx); err != nil {
+				workCancel(err)
```

Comment:
Hmm, spurious diff perhaps, although upon reviewing it I'm not sure whether the "right" behavior is to cancel the work context before or after stopping all producers. I'm thinking maybe before because something has obviously gone wrong on startup and we should bail out sooner rather than letting long-running jobs keep going indefinitely?